src/main/c/cext/ruby.c

```diff
@@ -1396,6 +1396,15 @@ VALUE rb_enc_str_new(const char *ptr, long len, rb_encoding *enc) {
   return truffle_invoke(rb_str_new(ptr, len), "force_encoding", rb_enc_from_encoding(enc));
 }
 
+void rb_enc_raise(rb_encoding *enc, VALUE exc, const char *fmt, ...) {
+    va_list args;
+    VALUE mesg;
+    va_start(args, fmt);
+    mesg = (VALUE) truffle_invoke(rb_mKernel, "sprintf", rb_str_new_cstr(fmt), args);
```

Comment:
Does this `va_list` form work? Are the extra arguments passed as an array or they are splat to truffle_invoke?

Reply:
@eregon It wasn't working with more than 1 argument so I've pushed an update to use a function with a `va_list` argument.